### PR TITLE
Extend service struct

### PR DESCRIPTION
**Pull request by Poltergeist**
Extend the service struct with a couple of get functions, those are needed for the deploy a service modal.

Also introducing a `ServiceUtil.js` this contains for now 3 functions:
- `convertFormModelToService` - This function is creating a `Service` object from the a `formModel` which is provided by the `AdvancedConfig` component.
- `createFormModelFromSchema` - This is creating a `formModel` with a given `Schema` and `Service`.
- `getAppDefinitionFromService` - This is creating a marathon `appDefinition` from a `Service` Object. 

This is a dependency for #212

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @pierlo-upitup

File: src/js/utils/ServiceUtil.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+const ServiceUtil = {
+  convertFormModelToService: function (formModel) {
+    formModel = Object.keys(formModel).reduce(function (memory, section) {
+      Object.keys(formModel[section]).forEach(function (attribute) {
+        memory[attribute] = formModel[section][attribute];
+      });
+      return memory;
+    }, {});
+    return new Service(formModel);
```
Comment (jfurrow) [inline src/js/utils/ServiceUtil.js]: Could we add a newline above this `return`, and the other `returns` above this?

**Comment by jfurrow** [comment]
Looks great, just one minor nit-pick comment!

File: src/js/utils/ServiceUtil.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import Service from '../structs/Service';
+
+const getFindPropertiesRecursive = function (service) {
```
Comment (orlandohohmeier) [inline src/js/utils/ServiceUtil.js]: Is there any special reason you curry this? Could be a plain function with to params...

File: src/js/utils/ServiceUtil.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      memo[subItem] = service.get(subItem) || item[subItem].default;
+      return memo;
+    }, {});
+  };
+  return findPropertiesRecursive;
+};
+
+const ServiceUtil = {
+  convertFormModelToService: function (formModel) {
+    formModel = Object.keys(formModel).reduce(function (memory, section) {
```
Comment (pierlo-upitup) [inline src/js/utils/ServiceUtil.js]: `memory`? How about the classic `memo`?

Comment (pierlo-upitup) [inline src/js/utils/ServiceUtil.js]: Also why are you mutating `formModel` directly? I'd rather have

```
convertFormModelToService: function (formModel) {
  let values = Object.keys(formModel)...
  // ...
  return new Service(values);
```

File: src/js/utils/ServiceUtil.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+      memo[subItem] = service.get(subItem) || item[subItem].default;
+      return memo;
+    }, {});
+  };
+  return findPropertiesRecursive;
+};
+
+const ServiceUtil = {
+  convertFormModelToService: function (formModel) {
```
Comment (pierlo-upitup) [inline src/js/utils/ServiceUtil.js]: `createServiceFromForm` ? #namingishardmyfriend

**Comment by pierlo-upitup** [comment]
Once comments are address ⛵

File: src/js/utils/__tests__/ServiceUtil-test.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+          cmd: 'sleep 1000;'
+        }
+      };
+
+      let expectedService = new Service({
+        id: '/test',
+        cmd: 'sleep 1000;'
+      });
+
+      expect(ServiceUtil.createServiceFromFormModel(model)).toEqual(expectedService);
```
Comment (pierlo-upitup) [inline src/js/utils/__tests__/ServiceUtil-test.js]: please newline `.toEqual` as the line doesn't fit

**Comment by pierlo-upitup** [comment]
One last nitpick and then 🚢 !

**Pull request merged**